src/hotspot/share/runtime/perfMemory.hpp

```diff
@@ -134,7 +134,7 @@ class PerfMemory : AllStatic {
     static size_t used() { return (size_t) (_top - _start); }
     static size_t capacity() { return _capacity; }
     static bool is_initialized();
-    static bool is_destroyed() { return _destroyed; }
+    static bool is_destroyed();
```

Comment:
So don't we need to make changes here? I think we can keep current implementation ( `is_destroyed()` )

Reply:
I was also partly going for consistency with _initialized and _destroyed w.r.t. int vs bool and 0/1 vs false/true. I can go back to making _destroyed all bool.

Reply:
Update done. I should also point out that _destroyed is not exported from VMStructs, so could have been made volatile long ago without needing to add any VMStructs support. SA only looks at _initialized, but I suppose you could argue that it also should be checking _destroyed before accessing PerfMemory.

Reply:
> SA only looks at _initialized, but I suppose you could argue that it also should be checking _destroyed before accessing PerfMemory.

In JDK-8151815, we do not clear fields in `PerfMemory` because they are used in SA even if they are freed, so we can ignore `_destroyed` in SA. We've discussed about this in [review thread of JDK-8151815](https://mail.openjdk.org/pipermail/serviceability-dev/2017-October/022050.html).